src/cache-distributions/poetry-cache.ts

```diff
@@ -0,0 +1,76 @@
+import * as glob from '@actions/glob';
+import * as os from 'os';
+import * as path from 'path';
+import * as exec from '@actions/exec';
+
+import CacheDistributor from './cache-distributor';
+
+class PoetryCache extends CacheDistributor {
+  constructor(
+    private pythonVersion: string,
+    protected patterns: string = '**/poetry.lock'
+  ) {
+    super('poetry', patterns);
+  }
+
+  protected async getCacheGlobalDirectories() {
+    const poetryConfig = await this.getPoetryConfiguration();
+
+    const cacheDir = poetryConfig['cache-dir'];
+    const virtualenvsPath = poetryConfig['virtualenvs.path'].replace(
+      '{cache-dir}',
+      cacheDir
+    );
+
+    const paths = [virtualenvsPath];
+
+    if (poetryConfig['virtualenvs.in-project'] === true) {
+      paths.push(path.join(process.cwd(), '.venv'));
+    }
+
+    return paths;
+  }
+
+  protected async computeKeys() {
+    const hash = await glob.hashFiles(this.patterns);
+    const primaryKey = `${this.CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-python-${this.pythonVersion}-${this.packageManager}-${hash}`;
+    const restoreKey = undefined;
+    return {
+      primaryKey,
+      restoreKey
+    };
+  }
+
+  private async getPoetryConfiguration() {
+    const {stdout, stderr, exitCode} = await exec.getExecOutput('poetry', [
+      'config',
+      '--list'
+    ]);
+
+    if (exitCode && stderr) {
+      throw new Error(
+        'Could not get cache folder path for poetry package manager'
+      );
+    }
+
+    const lines = stdout.trim().split(os.EOL);
```

Comment:
Tests on Windows are failing.  If I had to guess, the file `__tests/data/poetry.lock` is checked in with `\n` line endings and Git is not converting line endings on checkout in the CI run, so this is not splitting lines correctly on Windows.

I think this will work in any case:

```suggestion
    const lines = stdout.trim().split('\n');
```

If that doesn't fix the test and you don't have access to a Windows dev environment, adding some print line debugging could help.

Reply:
uh thanks! sorry I didn't notice the windows tests were failing, I'll keep an eye on them!